components/EditSPLToken.tsx

```diff
@@ -0,0 +1,176 @@
+import { FormikErrors, useFormik } from 'formik'
+import {
+  clusterApiUrl,
+  Connection,
+  PublicKey,
+  Transaction,
+} from '@solana/web3.js'
+import { FanoutClient, Wallet } from '@glasseaters/hydra-sdk'
+import { useAnchorWallet } from '@solana/wallet-adapter-react'
+import { useRef, useState } from 'react'
+import { isValidPubKey } from '../utils/utils'
+import {useAppSelector} from "../hooks/useAppSelector";
+import {selectCluster} from "../redux/features/wallet/walletSlice";
+
+interface FormValues {
+  acceptSPL: boolean
+  pubKeySPL: string
+}
+
+interface Props {
+  onCancel: Function
+  onSuccess: Function
+  hydraPubKey: string
+}
+
+const EditSPLToken = ({ onCancel, onSuccess, hydraPubKey }: Props) => {
+  let checkboxRef = useRef<HTMLInputElement>(null)
+  const cluster = useAppSelector(selectCluster)
+  const [loading, setLoading] = useState(false)
+
+  const connection = new Connection(clusterApiUrl(cluster), 'confirmed')
+
+  const wallet = useAnchorWallet() as Wallet
+  const fanoutSdk = new FanoutClient(connection, wallet)
+
+  const initialValues = {
+    acceptSPL: false,
+    pubKeySPL: '',
+  }
+
+  const tx = new Transaction()
+
+  const onSubmit = async (values: any) => {
+    setLoading(true)
+
+    try {
+      const ixSPL = await fanoutSdk.initializeFanoutForMintInstructions({
+        fanout: new PublicKey(hydraPubKey),
+        mint: new PublicKey(values.pubKeySPL),
+      })
+      tx.add(...ixSPL.instructions)
+
+      tx.recentBlockhash = (await connection.getLatestBlockhash()).blockhash
+      tx.feePayer = wallet.publicKey
+      const txSigned = await wallet.signTransaction(tx)
+
+      // Call the API to add spl token
+      const res = await fetch(`/api/addSplToken/${hydraPubKey}?cluster=${cluster}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          tx: txSigned.serialize().toString('base64'),
+          splToken: values.pubKeySPL,
+        }),
+      })
+
+      if(res.status == 200) {
+        onSuccess(values.pubKeySPL)
+      } else {
+        setLoading(false)
+        const error = await res.json()
+        formik.setFieldError('pubKeySPL', error)
```

Comment:
`error` is an object with `msg` and possibly `logs`, so we need:

```typescript
formik.setFieldError('pubKeySPL', error.msg)
```

Reply:
@abdelhakeem I might need to change all the `addSplToken` responses to include a `msg` field for consistency since the transaction error response is the only one that includes a `msg` field. Thanks for pointing that out, I'm on it